fmpq.h

```diff
@@ -477,6 +477,13 @@ FLINT_DLL void _fmpz_vector_push_back_zero(_fmpz_vector_t v);
 
 FLINT_DLL void _fmpz_vector_append_ui(_fmpz_vector_t v, const ulong * a, slong n);
 
+FMPQ_INLINE void _fmpz_vector_swap(_fmpz_vector_t a, _fmpz_vector_t b)
```

Comment:
This seems out of place. I assume you'll move it. Having said that, why is _fmpz_vector_append_ui in fmpq?

Reply:
Internally there simply has to be a resizable fmpz vector type. I can move it to wherever you think is appropriate.

Reply:
There is also a _fmpz_mat22 in fmpq

Reply:
Maybe into fmpz_vec?

Reply:
Is it possible to call these fmpz_vec given that the existing functions have underscores?

Reply:
there is a _fmpz_vec_... interface that just deals with raw pointers and no resizing.
The _fmpz_vector_t is resizable and has to be, and this type is of course not exposed in the official interface.
I am not sure if mixing _fmpz_vec and _fmpz_vector naming conventions is a good idea. For me, vectors should be resizable, and this was not a part of the design of _fmpz_vec.

Reply:
What I'm saying is that the functions were made underscore functions precisely because they don't do their own memory management. If they aren't vectors, then perhaps the name is wrong. What is wrong with fmpz_list_t?

Reply:
So you proposing the current `_fmpz_vector_t` be renamed `fmpz_list_t` or `_fmpz_list_t`? The name doesn't really matter as long as it is not `fmpz_vec_t` because that won't make sense. There is a slight problem in that the usage of `_fmpz_vector_t` in the `fmpq` module unfortunately has some extra bells and whistles attached so those would have to be separated from the `fmpz_list_t`.

Reply:
taking these bells and whistles off would probably be a good idea.

Reply:
I think fmpz_list_t works. Since it does its own memory management it should be a non-underscore function. So long as you think it is acting as a kind of list that can grow, that seems like a better name than vector, which implies a (fixed) basis.

Reply:
If it's for internal use, maybe just recycle fmpz_poly_t? You could typedef fmpz_vec_t to fmpz_poly_t (to have a clearer type name), and use the same memory management and scalar functions (optionally with fmpz_vec_X aliases for the fmpz_poly_X functions).

Reply:
And where would this fmpz_list_t reside?

Reply:
I guess it would have to be its own module. But @fredrik-johansson 's idea isn't bad. I'd #define it per file rather than typedef it, which could cause problems in future. But if that would work, I'd also support that.

Reply:
If I `#typedef fmpz_poly_t fmpz_list_t`, in which header file should this line of code reside?

Reply:
You can do it in fmpz_poly.h if you like. I think if you add an fmpz_list.h just for that it will break the build system.